### PR TITLE
[ingress-nginx] proxy-failover switch to distroless

### PR DESCRIPTION
**Pull request by aleksey-su**
## Description
proxy-failover switch to distroless and nginx is now statically builded. New location a binary file /opt/nginx-static/sbin/nginx, new location a config file /opt/nginx-static/conf/nginx.conf

## Why do we need it, and what problem does it solve?
Unification of the approach to the development of module images

## Checklist
- [ ] The code is covered by unit tests.
- [x] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [x] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: ingress-nginx
type: feature
summary: Switch `proxy-failover` to use distroless image.
impact_level: default
```

### CODE REVIEW

**Comment by aleksey-su** [comment]
image
image
image

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @aleksey-su. I've started the workflow for label 'e2e/run/openstack'...

File: modules/402-ingress-nginx/images/proxy-failover/werf.inc.yaml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  to: /go/pkg
+shell:
+  install:
+  - cd /src
+  - GOPROXY={{ $.GOPROXY }} CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -ldflags='-extldflags "-static" -s -w' -o entrypoint main.go
+  - chown -R 64535:64535 entrypoint
+  - chmod 0700 entrypoint
+---
+image: {{ .ModuleName }}/{{ .ImageName }}
+from: {{ .Images.BASE_NGINX_ALPINE }}
```
Comment (RomanenkoDenys) [inline modules/402-ingress-nginx/images/proxy-failover/werf.inc.yaml:27]: it looks like we can build static nginx and use distroless

**Review by RomanenkoDenys — CHANGES_REQUESTED** [review]
use static build please

**Comment by aleksey-su** [comment]
image

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @aleksey-su. I've started the workflow for label 'e2e/run/openstack'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @aleksey-su. I've started the workflow for label 'e2e/run/openstack'...

**Comment by z9r5** [comment]
/e2e/abort openstack;Standard;containerd;1.25 6487781621 6487781621-1-con-1-25 /sys/deckhouse-oss/install:pr6028 debian@45.12.229.247

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @z9r5. I've recognized your '/e2e/abort' command and started the workflow...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @aleksey-su. I've started the workflow for label 'e2e/run/openstack'...

**Comment by aleksey-su** [comment]
befor migrations
image
image

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @aleksey-su. I've started the workflow for label 'e2e/run/openstack'...

**Review by RomanenkoDenys — APPROVED** [review]
LGTM

**Review by name212 — APPROVED** [review]

**Review by yalosev — APPROVED** [review]

**Review by nabokihms — APPROVED** [review]

**Pull request merged**